cmd/influxd/launcher/query_test.go

```diff
@@ -0,0 +1,217 @@
+		t.Fatal("failed call to write points")
+	}
+
+	rawQ := fmt.Sprintf(`from(bucket:"%s")
+	|> filter(fn: (r) => r._m ==  "cpu" and (r._f == "v1" or r._f == "v0"))
+	|> range(start:-1m)
+	`, be.Bucket.Name)
+
+	results := be.MustExecuteQuery(be.Org.ID, rawQ, be.Auth)
+	defer results.Done()
+	results.First(t).HasTablesWithCols([]int{4, 4, 5})
+}
+
+// This test initialises a default launcher writes some data,
+// and checks that the queried results contain the expected number of tables
+// and expected number of columns.
+func TestPipeline_WriteV2_Query(t *testing.T) {
+	t.Parallel()
+
+	be := RunLauncherOrFail(t, ctx)
+	be.SetupOrFail(t)
+	defer be.ShutdownOrFail(t, ctx)
+
+	// The default gateway instance inserts some values directly such that ID lookups seem to break,
+	// so go the roundabout way to insert things correctly.
+	req := be.MustNewHTTPRequest(
+		"POST",
+		fmt.Sprintf("/api/v2/write?org=%s&bucket=%s", be.Org.ID, be.Bucket.ID),
+		fmt.Sprintf("ctr n=1i %d", time.Now().UnixNano()),
+	)
+	phttp.SetToken(be.Auth.Token, req)
+
+	resp, err := nethttp.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	if resp.StatusCode != nethttp.StatusNoContent {
+		buf := new(bytes.Buffer)
+		if _, err := io.Copy(buf, resp.Body); err != nil {
+			t.Fatalf("Could not read body: %s", err)
+		}
+		t.Fatalf("exp status %d; got %d, body: %s", nethttp.StatusNoContent, resp.StatusCode, buf.String())
+	}
+
+	res := be.MustExecuteQuery(
+		be.Org.ID,
+		fmt.Sprintf(`from(bucket:"%s") |> range(start:-5m)`, be.Bucket.Name),
+		be.Auth)
+	defer res.Done()
+	res.HasTableCount(t, 1)
+}
+
+// QueryResult wraps a single flux.Result with some helper methods.
+type QueryResult struct {
+	t *testing.T
+	q flux.Result
+}
+
+// HasTableWithCols checks if the desired number of tables and columns exist,
+// ignoring any system columns.
+//
+// If the result is not as expected then the testing.T fails.
+func (r *QueryResult) HasTablesWithCols(want []int) {
+	r.t.Helper()
+
+	// _start, _stop, _time, _f
+	systemCols := 4
+	got := []int{}
+	if err := r.q.Tables().Do(func(b flux.Table) error {
+		got = append(got, len(b.Cols())-systemCols)
+		b.Do(func(c flux.ColReader) error { return nil })
+		return nil
+	}); err != nil {
+		r.t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		r.t.Fatalf("got %v, expected %v", got, want)
+	}
+}
+
+// TablesN returns the number of tables for the result.
+func (r *QueryResult) TablesN() int {
+	var total int
+	r.q.Tables().Do(func(b flux.Table) error {
+		total++
+		b.Do(func(c flux.ColReader) error { return nil })
+		return nil
+	})
+	return total
+}
+
+// MustExecuteQuery executes the provided query panicking if an error is encountered.
+// Callers of MustExecuteQuery must call Done on the returned QueryResults.
+func (p *Launcher) MustExecuteQuery(orgID platform.ID, query string, auth *platform.Authorization) *QueryResults {
```

Comment:
Just a note, adding methods to a type from a test package can cause some confusing errors later. The alternative would be to move the receiver into the argument list. But I can live with it for now.

Reply:
This type is different from the Launcher in launcher.go, it is a test wrapper around the other Launcher.